medusa/server/api/v2/base.py

```diff
@@ -90,13 +86,11 @@ def async_call(self, name, *args, **kwargs):
         try:
             method = getattr(self, 'http_' + name)
         except AttributeError:
-            raise HTTPError(405, '{name} method is not allowed'.format(name=name.upper()))
+            raise HTTPError(405)
 
         def blocking_call():
-            try:
-                return method(*args, **kwargs)
-            except Exception as error:
-                self._handle_request_exception(error)
+            result = self._check_authentication()
+            return method(*args, **kwargs) if result is None else result
```

Comment:
Your kinda asuming that the method call cant raise exceptions?

Reply:
I'll try to explain as best I can.
Basically - yes.
But the difference is the `RequestHandler` class has an exception handler, and we handle everything in `write_error` now.
https://github.com/pymedusa/Medusa/blob/91140e1694a2264affb964e9ffefb729018538f6/medusa/server/api/v2/base.py#L151-L165

`HTTPErrors` are handled as RESTful responses,
And anything other than `HTTPError` is handled as an actual exception and sends "Internal Server Error" as the response (or the traceback, if `app.DEVELOPER` is enabled, like before)

Reply:
>Your kinda asuming that the method call cant raise exceptions?

Sorry, I actually misread that.
Quite the opposite, I plan on it, but as I explained in the comment above, it's all being handled